packages/medusa/src/types/utils.ts

```diff
@@ -0,0 +1,8 @@
+/**
+ * Utility type used to remove some optional attributes (coming from K) from a type T
+ */
+export type WithRequiredProperty<T, K extends keyof T> = T &
```

Comment:
suggestion: Move that to common.ts since this the place where `PartialPick` and `Writable` are :)